### PR TITLE
US114169 Show alt text in grades dialog if user does not have create grades permissions

### PR DESCRIPTION
**Pull request by emil-furniss**

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-grades-dialog.js [hunk 1]
```diff
@@ -119,29 +122,35 @@ class ActivityGradesDialog extends ActivityEditorMixin(LocalizeMixin(RtlMixin(Mo
...
+								<div class="d2l-activity-grades-dialog-create-new-activity-name">${newGradeName}</div>
+								<div class="d2l-body-small">${scoreOutOf && !scoreOutOfError ? html`
+									${this.localize('points', { points: formatNumber(scoreOutOf, { maximumFractionDigits: 2 })})}
+								` : null }
+								</div>
 							</div>
 						</div>
-					</div>
+					` : html`
+						<div class="d2l-body-small">
```
Comment (wongvincent) [inline components/d2l-activity-editor/d2l-activity-grades-dialog.js]: I think this needs to be shown even when `this._createNewRadioChecked` is `false`. It should be shown when `canLinkNewGrade` is false and shouldn't care about the `createNewRadioChecked` value.

Comment (emil-furniss) [reply components/d2l-activity-editor/d2l-activity-grades-dialog.js]: There was some combination shenanigans between the different show/hide/select options that weren't properly accounted for. Fixed below.

File: components/d2l-activity-editor/d2l-activity-grades-dialog.js [hunk 2]
```diff
@@ -95,6 +107,10 @@ class ActivityGradesDialog extends ActivityEditorMixin(LocalizeMixin(RtlMixin(Mo
 		dialog.resize();
 	}
 
+	_onDialogOpen(e) {
+		e.target.resize();
```
[the remark this answers is not in the document]
Comment (dbatiste) [reply components/d2l-activity-editor/d2l-activity-grades-dialog.js]: Is there something the dialog should be handling here?  The dialog should size itself automatically when opened, and it supports async content as well.  Just want to make sure there isn't something that needs to be fixed in dialog code.

[the remark this answers is not in the document]
Comment (emil-furniss) [reply components/d2l-activity-editor/d2l-activity-grades-dialog.js]: No, I think the dialog is fine and this is down to user error. We've got async content here but are not using the dialog async capabilities properly. I'll merge this for now but go back in and refactor to fix up the async work.

[the remark this answers is not in the document]
Comment (dbatiste) [reply components/d2l-activity-editor/d2l-activity-grades-dialog.js]: Sounds good - let me know if you need anything from me.  :)

**Pull request merged**